app/services/idv/financials_step.rb

```diff
@@ -1,32 +1,44 @@
 module Idv
   class FinancialsStep < Step
-    def complete?
-      idv_form.finance_type && idv_session.financials_confirmation.try(:success?) ? true : false
+    def submit
+      if complete?
+        @success = true
+        idv_session.financials_confirmation = true
+        idv_session.params = idv_form.idv_params
+      else
+        @success = false
+        idv_session.financials_confirmation = false
+      end
+
+      result
     end
 
     private
 
-    def vendor_validate
-      result = vendor_validator.validate
-      idv_session.params = idv_form.idv_params if complete?
-      result
+    attr_reader :success
+
+    def complete?
+      form_valid? && vendor_validator.success?
     end
 
     def vendor_validator_class
       Idv::FinancialsValidator
     end
 
-    def analytics_event
-      Analytics::IDV_FINANCE_CONFIRMATION
-    end
-
     def vendor_errors
-      idv_session.financials_confirmation.try(:errors)
+      vendor_validator.errors if form_valid?
     end
 
     def vendor_params
       finance_type = idv_form.finance_type
       { finance_type => idv_form.idv_params[finance_type] }
     end
+
+    def result
```

Comment:
The Ruby Way seems to be a class for everything. Having a formal class to represent the return value of a-class-that-acts-like-a-Form isn't about the method-vs-hash-key argument, for me. It's about the formality of a data structure. We use formal classes everywhere in this app. Why is this hash special?

As far as 
> I would suggest we have the classes return an OpenStruct, rather than creating a new FormResponse type of class because that would mean creating a separate class to encapsulate the response for each form/service object because some need to return more than success and errors.

I'm not suggesting we need a special flavor of `FormResponse` for every kind of form. Just this:

```ruby
class FormResponse
  def initialize(success:, errors:, extra:)
    @success = success
    @errors = errors
    @extra = extra
  end

  def success?
    @success
  end

  def errors
    @errors
  end

  def to_h
    { success: @success, errors: @errors }.merge(@extra)
  end
end
```

and then the Analytics class can just call `to_h` on each FormResponse to serialize it.